### ISSUE TITLE
False positive ``arguments-differ`` when converting ``@staticmethod`` to/from bound method

### ISSUE THREAD

**Issue by tl24** [PyCQA/pylint#6019]
### Bug description

When changing a base class static method with the @staticmethod decorator to a bound method using self, the arguments-differ warning is being generated.  I don't believe this should be the case as calling the child class method using self works correctly.

```python
# pylint: disable=too-few-public-methods
class BaseClass:
    @staticmethod
    def handlers():
        return {}


class ChildClass(BaseClass):
    def handlers(self):
        return {"foo": self.foo_handler}

    def foo_handler(self):
        pass
```

### Command used

```shell
pylint tests/args_differ.py
```

### Pylint output

```shell
************* Module tests.args_differ
tests/args_differ.py:9:4: W0221: Number of parameters was 0 in 'BaseClass.handlers' and is now 1 in overridden 'ChildClass.handlers' method (arguments-differ)
```

### Expected behavior

No warnings generated

### Pylint version

```shell
pylint 2.13.2
astroid 2.11.2
Python 3.9.9 (main, Nov 15 2021, 18:05:17) 
[GCC 5.4.0 20160609]
```

### OS / Environment

Ubuntu 20.04

**Comment by cdce8p** [comment]
> I don't believe this should be the case as calling the child class method using self works correctly.

True, but the other way doesn't work. You can't replace a call to `Base.handlers()` with `Child.handlers()`. If you know that you'll be calling  the method with `self.xxx` anyway, it probably shouldn't be a static one.

_Btw. both pyright and mypy also emits an error here._
```
# pyright
Method "handlers" overrides class "BaseClass" in an incompatible manner
  Positional parameter count mismatch; base method has 0, but override has 1

# mypy
error: Signature of "handlers" incompatible with supertype "BaseClass"  [override]
note:      Superclass:
note:          @staticmethod
note:          def handlers() -> Dict[Any, Any]
note:      Subclass:
note:          def handlers(self) -> Dict[Any, Any]
```

**Comment by tl24** [comment]
Ok, so is the guidance here just to disable the "no-self-use" warning in the base class if the intent is for it to be overridden in the child class and possibly needing the self reference?

**Comment by cdce8p** [comment]
> Ok, so is the guidance here just to disable the "no-self-use" warning in the base class if the intent is for it to be overridden in the child class and possibly needing the self reference?

Exactly! IMO the `no-self-use` checker encourages the wrong behavior here. Just because the base implementation of a method doesn't use self, doesn't mean it should be a static method. I opened an issue for that a while ago
* https://github.com/PyCQA/pylint/issues/5502
